### ISSUE TITLE
New quest: board_type=* (for information=board)

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#1226]
### General
Affected tag(s) to be modified/added: [board_type](https://wiki.openstreetmap.org/wiki/Key:board_type)
Question asked: **What is this information board about?**

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose – it's quite useful to know, what you can see on a board
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one) – an information board always has to have a topic/must be about "something"
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary – you can best see what is written on it, if you stand in front of it
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort) – `information=board` is used 126 813 times, `board_type` only 72 308 times.

### Ideas for implementation

**Element selection:** I guess that is obvious, there are no special restrictions I can see

**Metadata needed:** no

**Proposed GUI:**
image picker?
However I think an image picker might not be best due to the fact that these boards can look very different. So maybe a simple text selection of the most common values (possibly with a small explanation) would be more suitable. You could also just grab the X most-used values from taginfo and  offer these for input. (possibly in translated versions via usual POEditor translations)

---

BTW, this would nicely complement https://github.com/westnordost/StreetComplete/issues/1115, as a possible follow-up quest.

**Comment by westnordost** [comment]
How to tag something that is about this region generally? (wildlife + plants + history + map + ...)

**Comment by rugk** [comment]
> How to tag something that is about this region generally? (wildlife + plants + history + map + ...)

Similar to cuisines a tag like "regional" could be used, IMHO. Or just the main topic, e.g. if you are in a nature park and have a board with natural info and a map on it, I'd still say it is "nature".

> What's the use case?

Seeing what is shown there when planing a trip or when you want to know whether it is worth to go there etc.
E.g. an information board with a map in a city center is totally different to one where nature stuff is shown.
Taginfo e.g. also shows it is used by http://www.historic.place/.

**Comment by westnordost** [comment]
True, ok. 
Though, the problem I see here is that it is somewhat of a free-form field and that it is sometimes difficult to clearly determine the "type" of a board. The former and the latter conflict with each other, because on the one hand, clear type definitions would be necessary, on the other hand there may be countless types (and subcategories of each other) that are yet to grow. A growing list of possible types will lead to the same situation as in `building` or `surface`, where early taggings like `paved`/`unpaved `or `residential`/`commercial` later turned into maybe-mostly-categories implicitly. You know how problematic this is for maintainability.
I think this list in the wiki is not well defined enough yet. Same problem as with the museum type, only worse, because a single info board may talk about all kind of specific things while a museum, I imagine, can at least be reasonably categorized somehow.

**Comment by matkoniecz** [comment]
Coding this is relatively easy. Main problem is research of tagging and later testing it.

@rugk - would you be interested in something like that

- preparing list of bord_type tagging that should cover all types (done by @rugk )
- in case of missing values - proposing and discussing them as new ones (like was done with say building=office - done by @rugk )
- coding test version of this quest (done by @matkoniecz )
- providing test version of StreetComplete (.apk file) (done by @matkoniecz )
- testing - would require tagging of about 100-200 information boards to check whatever quest is good enough (done by @rugk )
- graphics - maybe this can be done without them? Or maybe @westnordost is interested in doing that? Or maybe it will be possible to find some icons/photos?

In short: I do not think that this quest is worth the effort, but if someone will do entire tag research I am willing to do easy part and implement it.

But my queue of tag research and OSM discussions is extremely long and is not getting shorter. "improving bord_type=* tagging" task would be done around year 2300.

**Comment by westnordost** [comment]
I am always available to make an icon.

Thank you Mateusz to enumerate all what  is necessary to do for this beyond Java implementation. I think this is the situation for most proposed quest types that are not implemented yet.

**Comment by matkoniecz** [comment]
> I am always available to make an icon.

What about images for each board type in selection menu?

**Comment by rugk** [comment]
So this is a TODO? And nothing is "done" there, already. :smiley: (I already wondered while reading as I have not done anything of that [task list](https://help.github.com/articles/about-task-lists/))

> What about images for each board type in selection menu?

That should be relatively easy by combining [a board image](https://openclipart.org/detail/210203/misc-bag-notice-board) and some generic graphic of what is shown.
However, I would not even say images are needed here. But yeah…, likely it would be nicer.

> I think this is the situation for most proposed quest types that are not implemented yet.

Off-topic, but we could add such a tick list to the quest issue template…

---

> preparing list of bord_type tagging that should cover all types

Should be in the wiki, huh? Okay, maybe the order, but we can use taginfo for that. I think there are no real regional differences.
And if the wiki is not good (as indicated) then it needs to be fixed anyway. Obviously I also would not like to introduce new tags not already documented in the wiki/used by users.
But I can totally look on improving the wiki there, but I need to know what's missing.

**Comment by westnordost** [comment]
>What about images for each board type in selection menu?

For that not. It's too much effort for such an unimportant quest type.

**Comment by matkoniecz** [comment]
> Should be in the wiki, huh?

Is wiki already covering all reasonable cases, or is it like with buildings where many cases were missing and required improvements to documentation?

**Comment by rugk** [comment]
Okay, looked at it:
* We should include `board_type=board` in the query, as it is obviously a wrong/outdated and useless tagging
* In taginfo there is `board-type=map`, but we may use `tourism=information` + `information=map` for this, as recommend in the wiki.
* There is [`board_type=sight`, motsly in Europe/Germany](https://overpass-turbo.eu/s/CRk) not documented in the wiki. I think the tag makes sense, however, seeing the usage in Overpass.
* `board_type=welcome_sign`… yeah, why not?
* `board_type=information`makes not much sense for me, but is still used 318 times. Propbably kit is just a tagging error (with `tourism=information`)
* `sport` – possible
* `technology`– also okay, IMHO, also missing
* `hiking`– could be just map, IMHO; or nature depending on what is there
* `astronomy` – I guess that's really rare…
* `planet_walk` – only in Germany & Swiss, [this thing](https://de.wikipedia.org/wiki/Planetenweg), also rare and IMHO questionable to call this an (information) board
* `qr` – well, "information", but rare
* `art` – not documented, but could be useful
* `board_type=geography` – okay, too
* `name` – makes not really much sense, IMHO (if i9t's a person it could be just `history` or so

The other ones are used <50 times. 
But if we do not plan to use images, we can also just have an input field and then auto-complete to this list of X most used tags. Maybe "curated" a bit, to remove some wrong/useless values, but apart from that, the list of possibilities is not that long…

**Comment by matkoniecz** [comment]
OK, so next step is to document missing values on wiki.

And find set that is not overlapping (for example how one is supposed to tag board that is a map of old power station)?

**Comment by matkoniecz** [comment]
> OK, so next step is to document missing values on wiki.

> And find set that is not overlapping (for example how one is supposed to tag board that is a map of old power station)?

@rugk Are you planning to do that?

**Comment by rugk** [comment]
Okay, documented the values I considered to be useful in [the wiki](https://wiki.openstreetmap.org/wiki/Key:board_type#Values).

**Comment by westnordost** [comment]
I think only the common choices should be implemented, things below 0.5% on taginfo can be solved through leaving a note.

**Comment by matkoniecz** [comment]
I think that very rare but reasonable values can be skipped but also can be included, just not displayed by default (hidden behind "see more" button,  is implement already).

Or are you worrying about things other than interface overloading?

**Comment by westnordost** [comment]
> Or are you worrying about things other than interface overloading?

No

Am 20. November 2018 15:45:32 MEZ schrieb Mateusz Konieczny <notifications@github.com>:
>I think that very rare but reasonable values can be skipped but also
>can be included, just not displayed by default (hidden behind "see
>more" button,  is implement already).
>
>Or are you worrying about things other than interface overloading?

**Comment by matkoniecz** [comment]
Code is at https://github.com/westnordost/StreetComplete/compare/master...matkoniecz:board_type?expand=1

For now without photos or images

**Comment by matkoniecz** [comment]
@westnordost Are you considering icons also for this quest or should I (or @rugk) look for some photos that would match available selection and be readable at small size?

**Comment by westnordost** [comment]
Probably in this case, photos or icons would not be particularly useful. Perhaps best text-only.

**Comment by matkoniecz** [comment]
In that case it would probably make sense to make quest_parking_access.xml layout more generic as part of this PR.

On the other hand it is just a single RadioGroup and it may be not worth doing this.

**Comment by westnordost** [comment]
Sure, why not

**Comment by rugk** [comment]
> look for some photos that would match available selection and be readable at small size?

In the wiki you can see how that would look. I agree, photos of these signs are hardly useful.
However, I guess, some generic symbols that just hint the subject like 🍃 for nature or 🏰 for history etc. would be possible.

**Comment by matkoniecz** [comment]
@rugk Finally fully implemented :)

**Comment by rugk** [comment]
Great!, thanks. :smiley:

**Comment by westnordost** [comment]
Ah wait, I'll close this once the PR is merged
